youtube_dl/extractor/safari.py

```diff
@@ -238,3 +245,52 @@ def _real_extract(self, url):
         course_title = course_json['title']
 
         return self.playlist_result(entries, course_id, course_title)
+
+
+class SafariLearningPathIE(SafariBaseIE):
+    IE_NAME = 'safari:learning-path'
+    IE_DESC = 'safaribooksonline.com learning paths'
+
+    _VALID_URL = r'''(?x)
+                    https?://
+                        (?:
+                            (?:www\.)?safaribooksonline\.com/
```

Comment:
```suggestion
                            (?:www\.)?(?:safaribooksonline|learning\.oreilly)\.com/
```
This would handle also new URLs starting with learning.oreilly.